datalad/utils.py

```diff
@@ -776,8 +843,15 @@ def ensure_list_from_str(s, sep='\n'):
         return s
     return s.split(sep)
 
+@overload
+def ensure_dict_from_str(s: str, sep: str = '\n') -> Optional[dict[str, str]]:
+    ...
+
+@overload
+def ensure_dict_from_str(s: dict[K, V], sep: str = '\n') -> Optional[dict[K, V]]:
+    ...
 
-def ensure_dict_from_str(s, **kwargs):
+def ensure_dict_from_str(s: str | dict[K, V], sep: str = '\n') -> Optional[dict[str, str]] | Optional[dict[K, V]]:
```

Comment:
For similar reasons to above
```suggestion
def ensure_dict_from_str(s, sep='\n'):
```